templates/app/offline-worker.js

```diff
@@ -0,0 +1,126 @@
+/* Any copyright is dedicated to the Public Domain.
+ * http://creativecommons.org/publicdomain/zero/1.0/ */
+
+<% importScripts.forEach(function (filepath) {
+%>importScripts('<%- filepath %>');
+<% }); %>
+(function (self) {
+  'use strict';
+
+  // On install, cache resources and skip waiting so the worker won't
+  // wait for clients to be closed before becoming active.
+  self.addEventListener('install', function (event) {
+    event.waitUntil(oghliner.cacheResources().then(function () {
+      if (typeof self.skipWaiting === 'function') {
+        return self.skipWaiting();
+      }
+    }));
+  });
+
+  // On activation, delete old caches and start controlling the clients
+  // without waiting for them to reload.
+  self.addEventListener('activate', function (event) {
+    event.waitUntil(oghliner.clearOtherCaches().then(function () {
+      if (self.clients && typeof self.clients.claim === "function") {
+        return self.clients.claim();
+      }
+    }));
+  });
+
+  // Retrieves the request following oghliner strategy.
+  self.addEventListener('fetch', function (event) {
+    if (event.request.method === 'GET') {
+      event.respondWith(oghliner.get(event.request));
+    }
+    else {
+      event.respondWith(self.fetch(event.request));
+    }
+  });
+
+  var oghliner = self.oghliner = {
+
+    // This is the unique prefix for all the caches controlled by this worker.
+    CACHE_PREFIX: 'offline-cache:<%= cacheId %>:' + (self.registration ? self.registration.scope : '') + ':',
+
+    // This is the unique name for the cache controlled by this version of the worker.
+    get CACHE_NAME() {
+      return this.CACHE_PREFIX + '<%= cacheVersion %>';
+    },
+
+    // This is a list of resources that will be cached.
+    RESOURCES: [
+      '/',
+<% resources.forEach(function (pathAndHash) {
+%>      '<%- pathAndHash.path %>', // <%- pathAndHash.hash %>
+<% }); %>
+    ],
+
+    // Adds the resources to the cache controlled by this worker.
+    cacheResources: function () {
+      var _this = this;
+      var now = Date.now();
+      var baseUrl = self.location;
+      return _this.openCache()
+        .then(function (cache) {
```

Comment:
Nit: here too (and below) I would use the style we use elsewhere in this codebase, which is very similar to the style you use here, except that the `.then(…` dereferencing/invocation isn't indented:

```js
      return _this.openCache()
      .then(function (cache) {
```